File: producer/core/models/raw_data.py
from abc import ABC
from datetime import datetime

class RawData(ABC):
    def __init__(self,
                 *,
                 latitude: float,
                 longitude: float,
                 sensor_id: str,
                 timestamp: datetime = datetime.now(),
                 ) -> None:
        self.latitude = latitude
        self.longitude = longitude
        self.sensor_id = sensor_id
        self.timestamp = timestamp

    def __str__(self) -> str:
        return f'{self.__class__.__name__} {self.__dict__}'

File: flink/src/main/java/com/sevenlast/synccity/models/RawData.java
package com.sevenlast.synccity.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor(force = true)
public abstract class RawData {
    private String sensorUuid;
    private String sensorName;
    private String groupName;
    private double latitude;
    private double longitude;
    private LocalDateTime timestamp;
}
